stl/inc/inplace_vector

```diff
@@ -0,0 +1,1560 @@
+#endif // _ITERATOR_DEBUG_LEVEL == 2
+
+        _STD _Move_unchecked(_Whereptr + 1, _Oldlast, _Whereptr);
+        _STD destroy_at(_Oldlast - 1);
+        --_My_size;
+
+        return _Make_iterator(_Whereptr);
+    }
+
+    constexpr iterator erase(const_iterator _First, const_iterator _Last) noexcept(
+        is_nothrow_move_assignable_v<value_type>) /* strengthened */ {
+        const pointer _Firstptr = const_cast<pointer>(_First._Unwrapped());
+        const pointer _Lastptr  = const_cast<pointer>(_Last._Unwrapped());
+        const pointer _Oldlast  = _My_data + _My_size;
+
+#if _ITERATOR_DEBUG_LEVEL == 2
+        _STL_VERIFY(_First._Getcont() == this && _Last._Getcont() == this
+                        && _Firstptr >= _My_data && _Lastptr >= _Firstptr && _Oldlast >= _Lastptr,
+            "inplace_vector erase iterator outside range");
+#endif // _ITERATOR_DEBUG_LEVEL == 2
+
+        if (_Firstptr != _Lastptr) {
+            const pointer _Newlast = _STD _Move_unchecked(_Lastptr, _Oldlast, _Firstptr);
+            _STD _Destroy_range(_Newlast, _Oldlast);
+            _My_size = static_cast<size_type>(_Newlast - _My_data);
+        }
+
+        return _Make_iterator(_Firstptr);
+    }
+
+    constexpr void swap(inplace_vector& _Right) noexcept(
+        _N == 0 || (is_nothrow_swappable_v<_Ty> && is_nothrow_move_constructible_v<_Ty>) ) {
+        // first swap the common subrange
+        const size_type _Minsize = (_STD min)(_My_size, _Right._My_size);
+        _STD _Swap_ranges_unchecked(_My_data, _My_data + _Minsize, _Right._My_data);
+
+        if (_My_size == _Right._My_size) {
+            return;
+        }
+
+        // next move the remaining elements from the larger container
+        const bool _Fromleft = _My_size >= _Right._My_size;
+        inplace_vector& _Src = _Fromleft ? *this : _Right;
+        inplace_vector& _Dst = _Fromleft ? _Right : *this;
+
+        _STD uninitialized_move(_Src._My_data + _Minsize, _Src._My_data + _Src._My_size, _Dst._My_data + _Minsize);
+        _STD _Destroy_range(_Src._My_data + _Minsize, _Src._My_data + _Src._My_size);
+        _STD swap(_Src._My_size, _Dst._My_size);
+    }
+
+    constexpr void clear() noexcept {
+        _STD _Destroy_range(_My_data, _My_data + _My_size);
+        _My_size = 0;
+    }
+
+    _NODISCARD constexpr friend bool operator==(const inplace_vector& _Left, const inplace_vector& _Right) {
+        if (_Left._My_size != _Right._My_size) {
+            return false;
+        }
+
+        return _STD equal(_Left._Unchecked_begin(), _Left._Unchecked_end(), _Right._Unchecked_begin());
+    }
+
+    _NODISCARD constexpr friend _Synth_three_way_result<_Ty> operator<=>(
+        const inplace_vector& _Left, const inplace_vector& _Right) {
+        return _STD lexicographical_compare_three_way(_Left._Unchecked_begin(), _Left._Unchecked_end(),
+            _Right._Unchecked_begin(), _Right._Unchecked_end(), _Synth_three_way{});
+    }

```

Comment:
```suggestion
    _NODISCARD constexpr friend auto operator<=>(const inplace_vector& _Left, const inplace_vector& _Right)
        requires requires(const _Ty __t) {
            { __t < __t } -> _Boolean_testable;
        }
    {
        return _STD lexicographical_compare_three_way(_Left._Unchecked_begin(), _Left._Unchecked_end(),
            _Right._Unchecked_begin(), _Right._Unchecked_end(), _Synth_three_way{});
    }
```

The approved standard wording is defective, which has the same problem as that reported in LWG-4071.

See also https://github.com/gcc-mirror/gcc/commit/d86472a6f041ccf3d1be0cf6bb15d1e0ad8f6dbe.